### PR TITLE
Convert PollyTestClient project to target dotnet 7

### PR DESCRIPTION
**Pull request by peter-csala**
# Description
- Upgraded `PollyTestClient` to target .NET 7
- Made necessary code changes to be able to compile
  - The project was referencing to Polly v5 ...

### CODE REVIEW

**Comment by martincostello** [comment]
If it didn't compile, that suggests to be it's missing from CI and needs adding to the solution file? Otherwise I'd have expected the CI to fail and have highlighted this previously.

File: PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public static async Task ExecuteAsync(CancellationToken cancellationToken)
         {
             Console.WriteLine(typeof(AsyncDemo09_Wrap_Fallback_Timeout_WaitAndRetry).Name);
             Console.WriteLine("=======");
-            // Let's call a web api service to make repeated requests to a server. 
+            // Let's call a web api service to make repeated requests to a server.
```
Comment (martincostello) [inline PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs:32]: Let's do a pass on the comments for this too and delete all the cruft and/or fix the wording like in the other samples.

**Comment by peter-csala** [comment]
> If it didn't compile, that suggests to be it's missing from CI and needs adding to the solution file? Otherwise I'd have expected the CI to fail and have highlighted this previously.

There were some breaking changes between Polly v5 and v7. In case of V5 you could write this code:
```cs
PolicyWrap policyWrap = Policy.WrapAsync(waitAndRetryPolicy, circuitBreakerPolicy);
```
In case of V7 you had to use `AsyncPolicyWrap` data type for the `policyWrap` variable. 
I've used `var` instead of prefixing everything with `Async` because we migrate this code anyway.
___
Long story short: CI did not fail because in V5 the code was valid. I had to fix it because in V7 that's not valid anymore.

**Comment by martincostello** [comment]
Ah I see - I read your description as:

- Use .NET 7
- Fix the project

Rather than:

- Update to .NET 7, which neccessitates various changes to work

File: PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs (hunk 1 above)
Comment (peter-csala) [reply PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs:32]: In this PR I've just migrated the project file and made it compile.

I will migrate the codebase gradually as with the previous project. At the end I'm planning to have a cleanup PR.

Does it make sense?

Comment (martincostello) [reply PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs:32]: It does, but at the same time I don't think the chunking up is particularly valuable here in terms of Git commit history etc. for some samples.

If we're touching some files and spot something that's pretty trivial to do, then I would have thought it easier to just change it now.

Comment (peter-csala) [reply PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs:32]: So, in this particular case what do you suggest what change should be included in this PR for this file?

Comment (martincostello) [reply PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs:32]: I would apply the same edits to the comments you've already made in the other samples (which you can just copy-paste in, right?) to the couple that are in this PR.

Comment (peter-csala) [reply PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs:32]: Ohh okay, I will do it

Comment (peter-csala) [reply PollyTestClient/Samples/AsyncDemo09_Wrap-Fallback-Timeout-WaitAndRetry.cs]: @martincostello Done. :)

**Review by martincostello — APPROVED** [review]

**Pull request merged**